File: Java/위장.java
package coding_test;

import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap <String, Integer> map = new HashMap<String, Integer>();
        
        for (int i = 0; i < clothes.length; i ++) {
            if (map.containsKey(clothes[i][1])) {
                map.put(clothes[i][1], map.get(clothes[i][1]) + 1);
            } else {
                map.put(clothes[i][1], 1);
            }
        }
        
        for (String key : map.keySet()) {
            answer *= map.get(key) + 1;
        }
        
        return answer - 1;
    }
}

File: JavaScript/위장.js
function solution(clothes) {
    let answer = 1;
    let dict = {};

    for (var i in clothes) {
        let tmp = clothes[i][1];
        if (tmp in dict) {
            dict[tmp] += 1;
        } else {
            dict[tmp] = 1;
        }
    }

    for (var i in dict) {
        answer *= dict[i] + 1
    }

    return answer - 1;
}